src/calendar/day/period/index.tsx

```diff
@@ -1,247 +1,197 @@
 import XDate from 'xdate';
-import isEqual from 'lodash/isEqual';
 import PropTypes from 'prop-types';
-
-import React, {Component} from 'react';
-import {TouchableWithoutFeedback, Text, View, ViewStyle} from 'react-native';
+import React, {useCallback, useRef, useMemo} from 'react';
+import {TouchableWithoutFeedback, TouchableOpacity, Text, View, ViewStyle, ViewProps, TextStyle, StyleProp} from 'react-native';
 
 import {xdateToData} from '../../../interface';
 import {Theme, DayState, DateData} from '../../../types';
-import {shouldUpdate} from '../../../componentUpdater';
-import * as defaultStyle from '../../../style';
 import styleConstructor from './style';
 import Dot from '../dot';
 import {MarkingProps} from '../marking';
 
 
-interface PeriodDayProps {
-  state?: DayState;
-  marking?: MarkingProps;
+export interface PeriodDayProps extends ViewProps {
   theme?: Theme;
+  date?: string;
+  marking?: MarkingProps;
+  state?: DayState;
   onPress?: (date?: DateData) => void;
   onLongPress?: (date?: DateData) => void;
-  date?: string;
   accessibilityLabel?: string;
   testID?: string;
 }
 
-export default class PeriodDay extends Component<PeriodDayProps> {
-  static displayName = 'PeriodDay';
-
-  static propTypes = {
-    state: PropTypes.oneOf(['selected', 'disabled', 'inactive', 'today', '']),
-    marking: PropTypes.any,
-    theme: PropTypes.object,
-    onPress: PropTypes.func,
-    onLongPress: PropTypes.func,
-    date: PropTypes.string
-  };
-
-  theme: Theme;
-  style: any;
-  markingStyle: any;
-  dateData: any;
-
-  constructor(props: PeriodDayProps) {
-    super(props);
-
-    this.theme = {...defaultStyle, ...(props.theme || {})};
-    this.style = styleConstructor(props.theme);
-    this.markingStyle = this.getDrawingStyle(props.marking);
-    this.dateData = props.date ? xdateToData(new XDate(props.date)) : undefined;
-  }
-
-  onPress = () => {
-    this.props.onPress?.(this.dateData);
-  };
-
-  onLongPress = () => {
-    this.props.onLongPress?.(this.dateData);
-  };
-
-  shouldComponentUpdate(nextProps: PeriodDayProps) {
-    const newMarkingStyle = this.getDrawingStyle(nextProps.marking);
-    if (!isEqual(this.markingStyle, newMarkingStyle)) {
-      this.markingStyle = newMarkingStyle;
-      return true;
-    }
+type MarkingStyle = {
+  containerStyle: StyleProp<ViewStyle>;
+  textStyle: StyleProp<TextStyle>;
+  startingDay?: ViewStyle;
+  endingDay?: ViewStyle;
+  day?: ViewStyle;
+}
 
-    return shouldUpdate(this.props, nextProps, ['children', 'state', 'marking', 'onPress', 'onLongPress', 'date']);
-  }
+const PeriodDay = (props: PeriodDayProps) => {
+  const {theme, marking, date, onPress, onLongPress, state, accessibilityLabel, testID, children} = props;
+  const dateData = date ? xdateToData(new XDate(date)) : undefined;
```

Comment:
I noticed this a few times, where you check if a value is undefined before passing it to the a util function. 
Consider moving this logic to the util and return undefined if the argument passed is undefined. 
It's very similar to how lodash works (they know how to handle undefined parameters) 
Also, it makes the code a little cleaner

Reply:
Changing the util to return undefined means to change the API for callback props that return this object. We can do it in V2